File: module-load-balance-gateway/src/main/java/com/xmmufan/permission/service/UserGroupService.java
package com.xmmufan.permission.service;

import com.google.protobuf.ServiceException;
import com.xmmufan.permission.domain.rbac.Permission;
import com.xmmufan.permission.domain.rbac.User;
import com.xmmufan.permission.domain.rbac.UserGroup;

import java.util.List;

/**
 * @author 黄源钦
 * @Description
 * @Date 16:13
 */
public interface UserGroupService {

    void createUserGroup(UserGroup group);

    void updateUserGroup(UserGroup group);

    void disableUserGroup(String groupId);

    void enableUserGroup(String groupId);

    void grantPermission(String groupId, List<Permission> grants);

    void revokePermission(String groupId, List<Permission> revokes);

    void addUserToGroup(String groupId ,String userId);

    void removeUserFromGroup(String groupId, String userId) throws ServiceException;


}

File: module-load-balance-gateway/src/main/java/com/xmmufan/permission/service/UserGroupService.kt
package com.xmmufan.permission.service

import com.xmmufan.permission.domain.rbac.UserGroup

/**
 * @author 黄源钦
 * @Description
 * @Date 16:13
 */
interface UserGroupService {

    fun createUserGroup(group: UserGroup)

    fun disableUserGroup(groupId: String)

    fun enableUserGroup(groupId: String)

    fun updatePermission(groupId: String, modifyIds: List<String>)

    fun updateGroupUser(groupId: String, userIds: List<String>)

}
